subprojects/language-jvm/src/main/java/org/gradle/api/internal/tasks/compile/DefaultJvmLanguageCompileSpec.java

```diff
@@ -63,12 +65,12 @@ public void setTempDir(File tempDir) {
 
     @Override
     public FileCollection getSource() {
-        return source;
+        return ImmutableFileCollection.of(source);
     }
 
     @Override
     public void setSource(FileCollection source) {
-        this.source = source;
+        this.source = source.getFiles();
```

Comment:
Looking at this I think it might be best to avoid even referring to `FileCollection` in this class. We should instead pass a `Set<File>` and make a defensive copy via `ImmutableSet.copyOf()`.

Reply:
It would perhaps be a good idea to standardize on the same collection type: we have `Iterable<File>`, `FileCollection` and `List<File>` used in this class already. I'd say `Iterable<File>` or `Collection<File>` should be fine for the setters. We could use the same for the getters, or be more specific, and return `ImmutableSet`. The goal should be consistency, though.

Reply:
Commit b2e3ea6408d swaps in `Iterable<File>` (& `Set<File>`) for `FileCollection`.  There are some ancillary changes necessary because `FileCollection` (& `FileTree`) was used to filter out files.  Let me know if think this change is worth keeping/pursuing.